src/main/application-update/quit-and-install-update.injectable.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import electronQuitAndInstallUpdateInjectable from "../electron-app/features/electron-quit-and-install-update.injectable";
+import { getCurrentDateTime } from "../../common/utils/date/get-current-date-time";
+import emitEventInjectable from "../../common/app-event-bus/emit-event.injectable";
+import discoveredUpdateVersionInjectable from "../../common/application-update/discovered-update-version/discovered-update-version.injectable";
+
+const quitAndInstallUpdateInjectable = getInjectable({
+  id: "quit-and-install-update",
+
+  instantiate: (di) => {
+    const electronQuitAndInstallUpdate = di.inject(
+      electronQuitAndInstallUpdateInjectable,
+    );
+
+    const emitEvent = di.inject(emitEventInjectable);
+    const discoveredUpdateVersion = di.inject(discoveredUpdateVersionInjectable);
+
+    return () => {
+      const discoveredVersion = discoveredUpdateVersion.value.get();
+
+      if (!discoveredVersion) {
+        throw new Error("Tried to install update but no update was discovered.");
+      }
+
+      emitEvent({
+        name: "app",
+        action: "start-installing-update",
+
+        params: {
+          version: discoveredVersion.version,
+          updateChannel: discoveredVersion.updateChannel.id,
+          currentDateTime: getCurrentDateTime(),
+        },
+      });
```

Comment:
Common sense. Things that I consider (not in any particular order):

1. Am I able to keep usage cleaner by giving name for it? (-> function in same file)
2. Does it have it's own dependencies? (-> injectable)
3. Is it used in multiple places? (-> injectable if there is dependencies, otherwise function)
4. Is it something that need to be overridden? (-> injectable)
5. Does it cause side effects? (-> injectable)

So, I would definitely not create injectable just for sake of injectable. There needs to be reason for it. 

`emitCurrentVersionToAnalyticsInjectable` is because it's `runnable`, which is ran when application is loaded.

Reply:
Okay those are roughly the criteria I would have followed too, though given that this might make usage clearer with its own name and it has some dependencies of its own I would have moved it to an injectable.

Minor though.